engine/proofview.ml

```diff
@@ -588,6 +588,13 @@ let shelve =
   InfoL.leaf (Info.Tactic (fun () -> Pp.str"shelve")) >>
   Shelf.modify (fun gls -> gls @ initial)
 
+let shelve_goals l =
+  let open Proof in
+  Comb.get >>= fun initial ->
+  let comb = CList.filter (fun g -> not (CList.mem g l)) initial in
+  Comb.set comb >>
+  InfoL.leaf (Info.Tactic (fun () -> Pp.str"shelve_goals")) >>
+  Shelf.modify (fun gls -> gls @ l)
```

Comment:
I don't know. But as per the documentation of `shelve_goals` you can give it goals which are in the focus context but not focused. After these last line, unless I'm mistaken, these goals will be in two places at once, which isn't supposed to happen.